From e48b70bea1ca8e22401ac9b9826224a930b895f6 Mon Sep 17 00:00:00 2001
From: Jan Kratochvil <jan.kratochvil@redhat.com>
Date: Mon, 27 Aug 2012 16:47:15 +0000
Subject: [PATCH] gdb/
 	* cli/cli-decode.c (print_doc_line): Keep skipping '.' and ',' not
 	followed by a whitespace.

---
 gdb/ChangeLog        |    5 +++++
 gdb/cli/cli-decode.c |    5 ++++-
 2 files changed, 9 insertions(+), 1 deletions(-)

Index: gdb-7.2/gdb/cli/cli-decode.c
===================================================================
--- gdb-7.2.orig/gdb/cli/cli-decode.c
+++ gdb-7.2/gdb/cli/cli-decode.c
@@ -1002,8 +1002,11 @@ print_doc_line (struct ui_file *stream, 
       line_buffer = (char *) xmalloc (line_size);
     }
 
+  /* Keep printing '.' or ',' not followed by a whitespace for embedded strings
+     like '.gdbinit'.  */
   p = str;
-  while (*p && *p != '\n' && *p != '.' && *p != ',')
+  while (*p && *p != '\n'
+	 && ((*p != '.' && *p != ',') || (p[1] && !isspace (p[1]))))
     p++;
   if (p - str > line_size - 1)
     {
